src/comments.py

```diff
@@ -0,0 +1,132 @@
+except ImportError:
+    from utils import EnvDefault, valid_repo_name, valid_token, valid_pr, valid_file, valid_sha, pick
+
+
+GITHUB_BASE = os.environ.get("GITHUB_BASE", "https://api.github.com")
+GITHUB_PR_COMMENTS_URL = GITHUB_BASE + "/repos/{repo_name}/pulls/{pull_request}/comments?access_token={token}"
+
+ACTIVE_CI = find_ci_provider()
+logging.basicConfig(level=logging.INFO)
+LOGGER = logging.getLogger(__name__)
+
+def format_filename(name):
+    root = gitroot()
+    for f in [os.path.join(root, name), name]:
+        full_name = os.path.abspath(f)
+        if os.path.exists(full_name):
+            return full_name[len(root)+1:]
+
+    return name
+
+def change_in_diff(filename, row, diff):
+    def file_in_diff(fn):
+        if filename == fn.path:
+            return True
+        elif os.path.join("a", filename) == fn.source_file:
+            return True
+        elif os.path.join("b", filename) == fn.target_file:
+            return True
+        return False
+
+    def line_in_hunk():
+        for ps in diff:
+            if file_in_diff(ps):
+                print(filename)
+                for hunk in ps:
+                    r = range(hunk.target_start, hunk.target_start + hunk.target_length + 1)
+                    if row in r:
+                        return True
+        return False
+
+    b = line_in_hunk()
+    if not b:
+        print(f"LINE: {row} NAME: {filename}")
+    return line_in_hunk()
+
+def update_comments(junit = None, repo_name = None, pull_request = None, sha = None, target_branch = "master", token = None):
+    url = GITHUB_PR_COMMENTS_URL.format(repo_name=repo_name, pull_request=pull_request, token=token)
+    headers = {"Content-Type": "application/json"}
+
+    print("piu")
+    diff = PatchSet(gitdiff(sha))
+
+    print("piu")
+
+    params = dict()
+    params['direction']='asc'
+    params['sort']='created'
+    result = requests.get(url, data=json.dumps(params), headers=headers)
+    body = json.loads(result.content.decode('utf8'))
+    existing_comments = []
+    for i in body:
+        existing_comments.append(pick(i, ["commit_id", "path", "position", "body"]))
+
+
+
+    print("ou")
+
+    payload  = dict()
+    matcher = r".*:\d{1,}:\d{1,}:\ .*"
+    junit_schema = XMLSchema("data/cistatus.xsd")
+    report = junit_schema.to_dict(junit)
+    for testsuite in report['testsuite']:
+        for testcase in testsuite['testcase']:
+            errors = testcase.get('error', False)
+            failures = testcase.get('failure', False)
+            filename = format_filename(testcase['@file'])
+            row = int(testcase['@line'])
+            if errors or failures and change_in_diff(filename, row, diff):
+                payload  = dict()
+                payload['commit_id'] = sha
+                payload['path'] = filename
+
+                payload['position'] = row
+                if errors:
+                    payload['body'] = testcase['error']['@message']
+                if failures:
```

Comment:
src/comments.py:105:5: F841 local variable 'final_diff' is assigned to but never used